Housing Card: Affordable Rental Units are Dwindling

DingoEatingFuzz opened the pull request:
Closes #56 

no tests 😞It's better to prioritize finishing cards at this point.

jaronheard commented:
@DingoEatingFuzz this is failing some tests in the component library. The tests could be updated, but it made me curious whether the changes to the label changed what the tooltip is displaying ? Do you know offhand?

DingoEatingFuzz commented:
@jaronheard I fixed all the things. My changes definitely busted the tooltip. I somehow didn't notice that I overrode the tooltip by using the VictoryBar label component for the labels. 

I brought back the tooltip and addressed the tests to reflect my changes. It's all sparkly again ✨

jaronheard commented on packages/2018-housing-affordability/src/components/AffordableRentalUnitsDwindling/index.js:
```diff
@@ -1,19 +1,134 @@
+const capitalize = str => str.length && str.split(' ')
+  .reduce((full, word) => `${full} ${word[0].toUpperCase() + word.substring(1)}`, '')
+  .trim();
 
```
you could also use `titleCase`, which I added to formatters in the component lib!

jaronheard commented on packages/2018-housing-affordability/src/components/AffordableRentalUnitsDwindling/index.js:
```diff
@@ -1,19 +1,134 @@
+            <p>
+              {capitalize(selectedCity)} ranks <strong className={emphasis}>{selectedCityLowRank.rank}/{selectedCityLowRank.total} </strong>
+              for new units that cost <strong className={emphasis}>&lt;$800/mo</strong>
+            </p>
```
@clemf gradient component goes here :)

jaronheard commented on packages/component-library/src/HorizontalBarChart/HorizontalBarChart.js:
```diff
@@ -50,19 +50,27 @@ const HorizontalBarChart = ({
+
+  const NegativeAwareTickLabel = props => (
+    <VictoryLabel dx={props.scale.y(minValue) - 20} {...props} textAnchor="end" />
+  );
```
Nice improvements for this component! 📊
